src/http/server/handlers/static_file_handler.cr

```diff
@@ -64,6 +64,14 @@ class HTTP::StaticFileHandler
       context.response.content_type = "text/html"
       directory_listing(context.response, request_path, file_path)
     elsif is_file
+      last_modified = File.stat(file_path).mtime.to_s("%a, %-d %h %C%y %T GMT")
+      if if_modified_since = context.request.headers.fetch("If-Modified-Since", nil)
+        if last_modified == if_modified_since
+          context.response.status_code = 304
+          return
+        end
+      end
+      context.response.headers["Last-Modified"] = last_modified
```

Comment:
The `Last-Modified` header should also be set if the response ist `304` as it is "useful for identifying or updating a previously cached response" (RFC 7232 section 3.3). Please just move this above the if branch.